### PR TITLE
add cmd to visualize iavl tree

### PR DESCRIPTION
**Pull request by yihuang**

### CODE REVIEW

File: iavl/visualize.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        n, _ = decode_node(db.get(prefix + node_key(hash)))
+        return n
+
+    def vis_node(hash: bytes, n: Node):
+        style = "solid" if n.version == version else "filled"
+        g.node(HexBytes(hash).hex(), label=label(node), style=style)
+
+    stack: List[bytes] = [root_hash]
+    while stack:
+        hash = stack.pop()
```
Comment (mmsqe) [inline iavl/visualize.py:34]: do we need check already expanded like [isinstance(hash, tuple)](https://github.com/crypto-com/python-iavl/blob/c0cea446ee9395656264a3cbdedc4b9dcad4bfc8/iavl/utils.py#L326)

File: iavl/cli.py [hunk 2]
```diff
@@ -285,5 +306,29 @@ def fast_rollback(
...
+@click.option(
+    "--version",
+    help="the version to query, default to latest version if not provided",
+    type=click.INT,
+)
+@click.argument("store")
+def visualize(db, version, store):
+    """
+    visualize iavl tree with dot, example:
+    $ iavl-cli virtualize --version 9 --db app.db bank | dot -Tpdf > /tmp/tree.pdf
```
Comment (mmsqe) [inline iavl/cli.py:322]:
```suggestion
    $ iavl-cli visualize --version 9 --db app.db bank | dot -Tpdf > /tmp/tree.pdf
```

File: iavl/visualize.py (hunk 1 above)
Comment (yihuang) [reply iavl/visualize.py:34]: no, that's for post-order traversal, we only do preorder here.

**Review by mmsqe — APPROVED** [review]

**Comment by yihuang** [comment]
[tree.pdf](https://github.com/crypto-com/python-iavl/files/10110592/tree.pdf)

Attached the iavl tree generated from the unit test db, with flag `--include-prev-version`.

File: iavl/cli.py [hunk 3]
```diff
@@ -285,5 +306,39 @@ def fast_rollback(
...
+    type=click.INT,
+)
+@click.option(
+    "--include-prev-version",
+    help="include the previous version to compare",
+    is_flag=True,
+    default=False,
+)
+@click.option("--store", "-s")
+def visualize(db, version, store=None, include_prev_version=False):
```
Comment (mmsqe) [inline iavl/cli.py:325]: optional store seems not correct

Comment (yihuang) [reply iavl/cli.py:325]: without passing store, it assume it's not a prefixed db, which is the case for the unit test db.

File: iavl/cli.py [hunk 4]
```diff
@@ -285,5 +306,39 @@ def fast_rollback(
...
+    """
+    visualize iavl tree with dot, example:
+    $ iavl-cli visualize --version 9 --db app.db bank | dot -Tpdf > /tmp/tree.pdf
```
Comment (mmsqe) [inline iavl/cli.py:328]:
```suggestion
    $ iavl-cli visualize --version 9 --db app.db --store bank | dot -Tpdf > /tmp/tree.pdf
```

**Review by mmsqe — APPROVED** [review]

**Pull request merged**